docs/feature_pointing_device.md

```diff
@@ -731,3 +732,507 @@ In general the following two functions must be implemented in appropriate locati
+| `pointing_mode.direction` | Direction based on stored x and y values  | `uint8_t`  |
+| `pointing_mode.x`         | Stored horizontal axis value              | `uint16_t` |
+| `pointing_mode.y`         | Stored vertical axis value                | `uint16_t` |
+   
+### Other tracked variables (Not directly accessible use functions)   
+These variables are tracked outside of the `pointing_mode_t` structure as they are not cleared on mode reset/change.
+
+|   Variable   | Description                                                     | Data type  | Functions 
+| :----------: | :-------------------------------------------------------------- | :--------: |
+| `tg_mode_id` | Mode id of last active toggle mode                              | `uint8_t`  |
+|  `is_left`   | if mode is on left pointing device (see split pointing below)   |   `bool`   |
+  
+***NOTE: `is_left` is only available when `SPLIT_POINTING_ENABLE` and `POINTING_DEVICE_COMBINED` are both defined***
+
+#### Controlling pointing device modes
+| Function                                   | Description                                                                                                                          | Return type |
+| :----------------------------------------- | :----------------------------------------------------------------------------------------------------------------------------------- | :---------: |
+| `set_pointing_mode_id(uint8_t mode_id)`    | Set current pointing device mode id to `mode_id`                                                                                     |   _None_    |
+| `toggle_pointing_mode_id(uint8_t mode_id)` | Set both stored toggle, and current mode id to `mode_id` <br /> or `POINTING_MODE_DEFAULT` if the stored toggle mode id is `mode_id` |   _None_    |
+| `get_pointing_mode_id(void)`               | Return current stored pointing device mode id                                                                                        |  `uint8_t`  |
+| `get_toggled_pointing_mode_id(void)`       | Return current stored toggle pointing device mode id                                                                                 |  `uint8_t`  |
+ 
+***Note: `toggle_pointing_mode_id(POINTING_MODE_DEFAULT)` can be used to clear the stored toggled mode but it will also reset the current active mode so that would need to be stored first and then reactivated if that is not the desired behaviour***
+ 
+These can be used to manipulate the current pointing mode id and toggle mode id at any time such as during layer changes, keypresses, tap dance sequences, and anywhere else in code (see examples below).
+
+#### Code example for changing modes on layer changes
+```c
+// in keymap.c
+// assuming enum and Layout for layers are already defined
+layer_state_t layer_state_set_user(layer_state_t state) {
+    // reset toggle to base pointing mode
+    toggle_pointing_mode_id(POINTING_MODE_DEFAULT);
```

Comment:
For the docs example on setting different modes per layer, wouldn't 
```C
if (get_toggled_pointing_mode_id() == !get_pointing_mode_id()) {
        toggle_pointing_mode_id(get_toggled_pointing_mode_id());}
````C=

more appropriate here and set_pointing_mode_id() be used in some of the examples to show? For example, I wanted PM_DRAG to persist (activated via keycode) but not PM_VOL, but PM_DRAG would always be reset with the example as written. Withe the suggested revisions, toggles are respected and momentary activations remain so

Reply:
I see we have very different needs/desires from switching pointing modes on layers.  I prefer to set the toggle mode by layer as that acts as a new "default" while on that layer so any change in momentary mode will go back to the layer default once deactivated until the layer changes. 

I'll add yours as a separate example with a brief explanation of the differences between the approaches.  One respects toggle from keys and the other does not with toggles intended to be activated strictly by layers.

Not sure I understand your suggested code snippet here though,  `get_toggled_pointing_mode_id() == !get_pointing_mode_id()`  will almost always return false resulting in the nested statement only executing when:
1. `tg_mode_id == PM_NONE && mode_id != PM_NONE` or
2. `tg_mode_id == PM_DRAG && mode_id == PM_NONE` 

With both simply resulting in both `tg_mode_id` and `mode_id` to being reset to `POINTING_MODE_DEFAULT` on execution of `toggle_pointing_mode_id(get_toggled_pointing_mode_id());`  is this your intended behaviour?

If I'm understanding what it is you would want from this based on your statement:   
1. Layer mode changes respect current toggled mode and do not overwrite/reset it
2. Toggled mode is reasserted on every layer change if different from current `mode_id`

Would the following code then meet your criteria? (if yes then I'll add it to the doc)
```c
layer_state_t layer_state_set_user(layer_state_t state) {
    // reassert toggle mode
    if (get_toggle_pointing_mode_id() != get_pointing_mode_id()) {
        set_pointing_mode_id(get_toggle_pointing_mode_id());
    }
    switch(get_highest_layer(state)) {
        case _NAVI:  // Navigation layer
            set_pointing_mode_id(PM_CARET);
            break;
        case _MEDIA: // Media control layer
            set_pointing_mode_id(PM_VOL);
            break;
    }
    return state;
}
```

Sorry if I'm not understanding correctly just let me know.

Thinking about your use case I realized there are some things I could improve about precision mode so thanks for that!